arch/arc/core/cache.c

```diff
@@ -25,6 +25,10 @@
 #include <init.h>
 #include <stdbool.h>
 
+#if defined(CONFIG_CACHE_LINE_SIZE_DETECT)
```

Comment:
Why moving this from already existing ifdef section for `CONFIG_CACHE_LINE_SIZE_DETECT` below?
IMHO less ifdefs makes code simpler and easier to digest.

Reply:
Because otherwise it fails to compile when enabling `ARCH_CACHE_FLUSH_DETECT`